plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/diagram/DefaultRequestTargetTypeHintsActionHandler.java

```diff
@@ -0,0 +1,43 @@
+/********************************************************************************
+ * Copyright (c) 2023 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.server.diagram;
+
+import java.util.List;
+
+import org.eclipse.glsp.server.actions.AbstractActionHandler;
+import org.eclipse.glsp.server.actions.Action;
+import org.eclipse.glsp.server.actions.CheckEdgeTargetResultAction;
+import org.eclipse.glsp.server.actions.RequestCheckEdgeTargetAction;
+
+import com.google.inject.Inject;
+
+/**
+ * Default handler implementation for {@link RequestCheckEdgeTargetAction}.
+ * This handler always rejects creation of dynamic Edges. Client applications
+ * should replace this default implementation with their own logic to
+ * conditionally accept creation of new edges configured with a Dynamic type.
+ */
+public class DefaultRequestTargetTypeHintsActionHandler extends AbstractActionHandler<RequestCheckEdgeTargetAction> {
```

Comment:
We typically omit the default prefix
```suggestion
public class RequestTargetTypeHintsActionHandler extends AbstractActionHandler<RequestCheckEdgeTargetAction> {
```